sdk/cosmosdb/cosmos/src/ClientContext.ts

```diff
@@ -600,7 +600,7 @@ export class ClientContext {
   }: {
```

Comment:
Can't we use an interface here? I just feel that defining method in this way impacts readability of this method.

Reply:
Can you please elaborate on this? Functions are `first class constructs` in JS ecosystem.

Reply:
I mean instead of using object definition here {body: T;
    path: string;
    partitionKey: string;
    partitionKey: PartitionKey;
    resourceId: string;
    options?: RequestOptions;
  }
  Should we create an interface and use it here? for better readability

Reply:
Sure, since this is unrelated to `hierarchical paritionkey` let me take this as a separate change. Tracking it here [partitionKey: PartitionKey;](https://github.com/Azure/azure-cosmosdb-node/issues/265)